### PR TITLE
refactor (request_)collection specs to use collection_shared_examples

### PR DESCRIPTION
**Pull request by ndushay**
## Why was this change made?

-    Reduce duplication of test code
-    Ensure Collection and RequestCollection models stay in sync.

Part of #70  (akin to PR #75, #76 for (Request)File and (Request)AdminPolicy models)

## Was the documentation (README, wiki) updated?

na

### CODE REVIEW

File: spec/cocina/models/request_collection_spec.rb [hunk 1]
```diff
@@ -1,228 +1,26 @@
...
-                                                                    titleFull: 'my collection')
+  it_behaves_like 'it has collection attributes'
 
-        link = collection.identification.catalogLinks.first
-        expect(link.catalog).to eq 'symphony'
-        expect(link.catalogRecordId).to eq '44444'
-      end
-    end
+  it 'does not have Checkable model methods' do
+    item = described_class.new(required_properties)
```
Comment (jcoyne) [inline spec/cocina/models/request_collection_spec.rb]: Do we need a test for something that is not present?  There are an infinite number of things this object does not do and they aren't important to test for.

Comment (ndushay) [reply spec/cocina/models/request_collection_spec.rb]: I can take it out;  I put it in to distinguish from Collection, but will remove.

**Comment by ndushay** [comment]
I have now removed `it 'does not have Checkable model methods' do` tests for RequestCollection, and added a commit to also remove the same for RequestFile and RequestAdminPolicy.

File: spec/cocina/models/collection_shared_examples.rb [hunk 2]
```diff
@@ -0,0 +1,290 @@
...
+              {
+                who: 'Other Justin',
+                what: 'self',
+                date: '2017-10-20T15:42:15Z',
+                to: 'Earthworks',
+                release: 'false'
+              }
+            ]
+          },
+          description: {
```
Comment (ndushay) [inline spec/cocina/models/collection_shared_examples.rb]: I removed sourceId attribute as it's not in the struct and I'm not sure we associate collection objects with a sourceId ?

**Pull request merged**